### ISSUE TITLE
Find an alternative for Alt keybindings for X/Linux users

### ISSUE THREAD

**Issue by badosu** [LMMS/lmms#1810]

**Comment by Sti2nd** [comment]
So the problem is that you cannot first press ALT and then use the mouse in Linux.

My thoughts: It would be really great to have just a single key because then it can be combined with fast resizing of notes in Piano Roll (and possibly other functionality ofc). Now in LMMS you can click and hold and while holding hit SHIFT to do fast resize, add the ALT key and you can do fine tuning quick too.

**Comment by mikobuntu** [comment]
I like the idea of just a single key i.e ALT , my preference would be to use the window management overrides of QT for Linux Xwindows. 
ALT has become or always has been a well known modifier that users have become accustomed to .

**Comment by Umcaruje** [comment]
>I like the idea of just a single key i.e keeping ALT intact as it is.... , my preference would be to use the window management overrides of QT for Linux Xwindows. 

I agree with mikobuntu, this would be an ideal solution to this problem (If its currently possible).

**Comment by tresf** [comment]
> preference would be to use the window management overrides

I'm not sure this can be controlled on the application level.  This is very wishful thinking.

From what I'm reading, you're dealing with global desktop settings here, not application-specific overrides.

I use ALT+Click quite a bit on Linux so I wouldn't be a fan of changing this default for the desktop personally.

**Comment by mikobuntu** [comment]
"I use ALT+Click quite a bit on Linux"

Yeah this will be the same for a lot of users and most likely that a user will be multi tasking, not just using LMMS. I think what @badosu had mentioned in the previous thread leading to this FR, was having a temp override, i.e only during the actual mouse/key gestures and not as I had initially proposed a global override.

**Comment by tresf** [comment]
>  was having a temp override

How though?  That is my question... is this even possible?

**Comment by Sti2nd** [comment]
If it is possible... Idk. I guess mikobuntu mean that in the Piano Roll when mouse over the green volume bars LMMS will override, but note elsewhere.

**Comment by mikobuntu** [comment]
@Sti2nd  yes that is what I mean and in any other situation where ALT key bindings are used within LMMS.  @tres " is this even possible? " IDK , hopefully :)

**Comment by mikobuntu** [comment]
http://qt-project.org/faq/answer/how_can_i_catch_altf4_in_my_qt_application

could something like this example work?

**Comment by musikBear** [comment]
dont want to say something thats to bleeding obvious, but it may be that shortcuts cant be defined 'cross' platformed. The huge work on the shortcut table, attempts to preserve the (little :) logic that existed, and to extend it -But will that be possible across all platforms? Im sure some conflicts (like the alt-linux) also will apear for mac / X, so perhaps its futile to attempt to make generalized cross-platforms shortcuts?

**Comment by Sti2nd** [comment]
Usually/always(?) the OS sticks with its special key, such as the Windows key (with logo), the Command key for Mac, and Super key for Linux, and I haven't experienced any problems with the shortcuts up until now. I admit I haven't tested too much on any platform yet... I think this is just a special case. If I find keys that are just as easy to use as Alt + click I would use them instead, but it might be that Linux have to be special in this case indeed.

**Comment by tresf** [comment]
> The huge work on the shortcut table, attempts to preserve the (little :) logic that existed, and to extend it -But will that be possible across all platforms?

Very valid points.... It will take some minor tweaking to make a shortcut table accurate for all platforms, but so far the ALT+Click is the only big one. :+1:
